src/open_dread_rando/misc_patches/elevator.py

```diff
@@ -1,15 +1,90 @@
+            if trans_type.default_icon_id == usable_id:
+                return trans_type
+        return None
+
+TRANSPORT_TYPES = {
+    "Elevator": TransporterIcon(
+        default_icon_id="UsableElevator",
+        coords=(2,4),
+        prefix="ELEVATOR TO ",
+        disabled_id="DisabledElevator"
+    ),
+    "Train": TransporterIcon(
+        default_icon_id="UsableTrain",
+        coords=(1,4),
+        prefix="SHUTTLE TO ",
+        disabled_id=''
+    ),
+    "Transport": TransporterIcon(
+        default_icon_id="UsableTransport",
+        coords=(8,2),
+        prefix = "TRANSPORT CAPSULE TO ",
+        disabled_id=''
+    )
+}
+def _patch_actor(editor: PatcherEditor, elevator: dict):
+    level = editor.get_scenario(elevator["teleporter"]["scenario"])
+    actor = level.actors_for_layer(elevator["teleporter"]["layer"])[elevator["teleporter"]["actor"]]
+    try:
+        usable = actor.pComponents.USABLE
+    except AttributeError:
+        raise ValueError(f'Actor {elevator["teleporter"]} is not a teleporter')
+    usable.sScenarioName = elevator["destination"]["scenario"]
+    usable.sTargetSpawnPoint = elevator["destination"]["actor"]
+
+def _patch_minimap_arrows(editor: PatcherEditor, elevator: dict):
+    map = editor.get_scenario_map(elevator["teleporter"]["scenario"])
+    sign = map.get_category("mapTransportSigns")[ elevator["teleporter"]["actor"] ]
+    sign["sDestAreaId"] = elevator["destination"]["scenario"]
+
+def _patch_map_icon(editor: PatcherEditor, elevator: dict):
+    # get transporter type from mapUsables entry
+    conn_name: str = elevator["connection_name"]
+    icon_id = f'RDV_TRANSPORT_{conn_name.replace(" ", "")}' # ex: "RDV_TRANSPORT_Artaria-ThermalDevice"
+
+    bmmap = editor.get_scenario_map(elevator["teleporter"]["scenario"])
+    usable = bmmap.get_category("mapUsables")[ elevator["teleporter"]["actor"] ]
+    ti = TransporterIcon.get_icon(usable)
+    if ti is None:
+        raise ValueError(f'Usable icon_id {usable["sIconId"]} invalid!')
+
+    # add BMMDEF
+    editor.map_icon_editor.add_icon(ti.map_icon(icon_id, elevator["connection_name"]))
 
+    # update usable
+    usable["sIconId"] = icon_id
 
 def patch_elevators(editor: PatcherEditor, elevators_config: list[dict]):
     for elevator in elevators_config:
         LOG.debug("Writing elevator from: %s", str(elevator["teleporter"]))
-        level = editor.get_scenario(elevator["teleporter"]["scenario"])
-        actor = level.actors_for_layer(elevator["teleporter"]["layer"])[elevator["teleporter"]["actor"]]
-        try:
-            usable = actor.pComponents.USABLE
-        except AttributeError:
-            raise ValueError(f'Actor {elevator["teleporter"]} is not a teleporter')
-        usable.sScenarioName = elevator["destination"]["scenario"]
-        usable.sTargetSpawnPoint = elevator["destination"]["actor"]
+        _patch_actor(editor, elevator)
+        _patch_minimap_arrows(editor, elevator)
```

Comment:
let's future proof this and avoid calling these functions if we're dealing with a Teleportal